src/modules/samples/cpp/src/lib/Sample.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <cstdio>
+#include <string>
+
+#include <CommonUtils.h>
+#include <Logging.h>
+
+#define SAMPLE_LOGFILE "/var/log/osconfig_sample.log"
+#define SAMPLE_ROLLEDLOGFILE "/var/log/osconfig_sample.bak"
+
+class SampleLog
+{
+public:
+    static OSCONFIG_LOG_HANDLE Get()
+    {
+        return m_log;
+    }
+
+    static void OpenLog()
+    {
+        m_log = ::OpenLog(SAMPLE_LOGFILE, SAMPLE_ROLLEDLOGFILE);
+    }
+
+    static void CloseLog()
+    {
+        ::CloseLog(&m_log);
+    }
+
+private:
+    static OSCONFIG_LOG_HANDLE m_log;
+};
+
+class Sample
+{
+public:
+    Sample(unsigned int maxPayloadSizeBytes);
+    virtual ~Sample() = default;
+
+    virtual int SetValue(const std::string& value);
+    virtual std::string GetValue();
```

Comment:
The static library class Get and Set methods should have same signatures as MmiGet and MmiSet respectively. See in modules.md:

[https://github.com/Azure/azure-osconfig/blob/main/docs/modules.md#121-module-static-library](https://github.com/Azure/azure-osconfig/blob/main/docs/modules.md#121-module-static-library)